python/wpt/grouping_formatter.py

```diff
@@ -112,14 +112,17 @@ class ServoHandler(mozlog.reader.LogHandler):
     """LogHandler designed to collect unexpected results for use by
     script or by the ServoFormatter output formatter."""
 
-    def __init__(self):
+    def __init__(self, flakes_detection=False):
+        """
+        Flake detection assumes first suite is actual run
+        and rest of the suites are retry-unexpected for flakes detection.
+        """
+        self.flakes_detection = flakes_detection
         self.reset_state()
 
     def reset_state(self):
-        self.number_of_tests = 0
-        self.completed_tests = 0
-        self.need_to_erase_last_line = False
-        self.running_tests: Dict[str, str] = {}
+        self.reset_stats()
```

Comment:
Let's do this to avoid confusion about `reset_state` and `reset_stats`, as the names look very similar:

```suggestion
    def reset_state(self):
        self.number_of_tests = 0
        self.completed_tests = 0
        self.need_to_erase_last_line = False
        self.running_tests: Dict[str, str] = {}
        if self.currently_detecting_flakes:
            return
        self.currently_detecting_flakes = False
```

Reply:
That wouldn't work because we will never set inital values for the rest, but I did rename `reset_stats` to `reset_statistics`.

Reply:
I think this will work right? `self.currently_detecting_flakes` is only set to true once the initial suite has run. What am I missing here?

Reply:
Ups, you are right.